src/main/java/com/faforever/client/chat/ChatUserService.java

```diff
@@ -38,114 +38,121 @@ public void afterPropertiesSet() {
   }
 
   public void populateClan(ChatChannelUser chatChannelUser) {
-    if (chatChannelUser.getClan().isEmpty()) {
-      chatChannelUser.getPlayer().ifPresent(player -> {
-        if (player.getClan() != null) {
-          clanService.getClanByTag(player.getClan())
-              .thenAccept(optionalClan -> Platform.runLater(() -> {
-                    if (optionalClan.isPresent()) {
-                      chatChannelUser.setClan(optionalClan.get());
-                      chatChannelUser.setClanTag(String.format("[%s]", optionalClan.get().getTag()));
-                    } else {
-                      chatChannelUser.setClan(null);
-                      chatChannelUser.setClanTag(null);
-                    }
-                  })
-              );
-        } else {
-          chatChannelUser.setClan(null);
-          chatChannelUser.setClanTag(null);
-        }
-      });
+    if (chatChannelUser.isDisplayed()) {
+      if (chatChannelUser.getClan().isEmpty()) {
+        chatChannelUser.getPlayer().ifPresent(player -> {
+          if (player.getClan() != null) {
+            clanService.getClanByTag(player.getClan())
+                .thenAccept(optionalClan -> Platform.runLater(() -> {
+                      if (optionalClan.isPresent()) {
+                        chatChannelUser.setClan(optionalClan.get());
+                      } else {
+                        chatChannelUser.setClan(null);
+                      }
+                    })
+                );
+          } else {
+            chatChannelUser.setClan(null);
+          }
+        });
+      }
+    } else {
+      chatChannelUser.setClan(null);
```

Comment:
Maybe we can refactor this since it is hardly readable anymore. Should be extracted into methods. Too many nesting